onnxruntime/core/graph/graph.cc

```diff
@@ -4062,7 +4062,13 @@ void Graph::ToGraphProtoInternal(ONNX_NAMESPACE::GraphProto& graph_proto) const
     *(graph_proto.mutable_output()->Add()) = output_arg->ToProto();
   }
 
-  for (const auto* value_info : value_info_) {
+  // Before adding value info to GraphProto, sort its unordered values so that across multiple calls
+  // to this function, its values are consistently written to the GraphProto
+  auto sort_predicate = [](const NodeArg* v1, const NodeArg* v2) { return v1->Name() < v2->Name(); };
+  std::vector<const NodeArg*> value_info_sorted{value_info_.begin(), value_info_.end()};
+  std::sort(value_info_sorted.begin(), value_info_set.end(), sort_predicate);
```

Comment:
Looks like `value_info_set` should actually be `value_info_sorted`

Reply:
My bad! Fixed it and verified build.